app/controllers/csv_validations_controller.rb

```diff
@@ -5,15 +5,30 @@ def create
       csv_text = uploaded_file.read
       institution = @navigation_context.institution
       csv_table = CSV.parse(csv_text, headers: true)
+
+      unless csv_table.headers == ["Batch", "Concentration", "Distractor", "Instructions"]
+        render json: { found_batches: [],
+                       not_found_batches: [],
+                       samples_nbr: 0, 
+                       error_message: "Invalid file format." }
+        return
+      end
+
       batch_values = csv_table.map { |row| row["Batch"] }
       found_batches = institution.batches.where(batch_number: batch_values).pluck(:batch_number)
       not_found_batches = (batch_values - found_batches).uniq
 
       render json: { found_batches: found_batches,
                      not_found_batches: not_found_batches,
-                     samples_nbr: csv_table.size }
+                     samples_nbr: csv_table.size,
+                     error_message: "" }
     else
-      render json: { error: "Invalid file format. Please upload a CSV file." }, status: :unprocessable_entity
+      unless csv_table.headers == ["Batch", "Concentration", "Distractor", "Instructions"]
+        render json: { found_batches: [],
+                       not_found_batches: [],
+                       samples_nbr: 0, 
+                       error_message: "Invalid file format." }
+      end
```

Comment:
This is identical to the above. Maybe extract a method for rendering the error json? and I believe we don't need to check the csv headers if the file isn't a text/csv file?

Reply:
>  and I believe we don't need to check the csv headers if the file isn't a text/csv file?
:facepalm: